### DISCUSSION TITLE
Block size and count on littlefs partition on ESP32-C3

### DISCUSSION THREAD

**Discussion by pablogventura** [micropython/micropython#12841] [ESP32] [answered]
Hello everyone, I'm working on building the filesystem externally in order to produce a bin file that includes the MicroPython build with the integrated filesystem. What would be the appropriate block size and block count for the LittleFS partition? Thanks in advance.

**Comment by rkompass** [comment]
Block size 256 seems to be a value that is good in most conditions (alternative block size 512 probably also quite good). Block count will depend then on how much space you will assign to the LittleFS partition afaiu.

**Comment by glenn20** [comment]
On ESP32 at least, the default block size for LFS is 4096 bytes.

**Comment by pablogventura** [comment]
I have downloaded the original fs created by MicroPython and I can read using littlefs-tools Python package on linux using block size 4096 and block count 512. Now, I am trying to create that partition from Linux and upload to ESP32-C3, and and MicroPython says:
```
The filesystem appears to be corrupted. If you had important data there, you
may want to make a flash snapshot to try to recover it. Otherwise, perform
factory reprogramming of MicroPython firmware (completely erase flash, followed
by firmware programming).
```
The comand to create littlefs image file I used is: `littlefs_create -b 4096 -c 512 -s fs_content/ -i fs.bin`
Command to upload partition: `esptool.py -p /dev/ttyUSB0 -b 460800 write_flash 2097152 fs.bin`

**Reply by pablogventura** [reply, marked as answer]
In ESP32-c3 block size is 4096 and block count is 512. I started a new topic with the question about creating the filesystem.

**Comment by Facuu13** [comment]
.
